File: app/src/main/java/com/unis/gameplatfrom/ui/PassWordFragment.java
package com.unis.gameplatfrom.ui;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.content.ContextCompat;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.blankj.utilcode.util.EmptyUtils;
import com.blankj.utilcode.util.EncryptUtils;
import com.blankj.utilcode.util.NetworkUtils;
import com.blankj.utilcode.util.RegexUtils;
import com.trello.rxlifecycle2.android.FragmentEvent;
import com.unis.gameplatfrom.R;
import com.unis.gameplatfrom.api.HUDLoadDataSubscriber;
import com.unis.gameplatfrom.api.PublicApiInterface;
import com.unis.gameplatfrom.api.RetrofitWrapper;
import com.unis.gameplatfrom.api.result.BaseObjectResult;
import com.unis.gameplatfrom.base.BaseFragment;
import com.unis.gameplatfrom.cache.UserCenter;
import com.unis.gameplatfrom.constant.LoginConstant;
import com.unis.gameplatfrom.entity.LoginEntity;
import com.unis.gameplatfrom.presenter.LoginPresenter;

import butterknife.BindView;
import butterknife.OnClick;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * 账号密码登陆
 */
public class PassWordFragment extends BaseFragment<LoginPresenter> implements LoginConstant.View {


    @BindView(R.id.txt_account)
    EditText mAccount;

    @BindView(R.id.txt_password)
    EditText mPassword;

    @BindView(R.id.btn_login_password)
    Button mLoginBtn;

    @BindView(R.id.save_account)
    CheckBox mSaveBtn;

    @BindView(R.id.save_text)
    TextView mSaveText;

    @BindView(R.id.click_layout)
    LinearLayout mClickLayout;

    private boolean saveaccount;

    @BindView(R.id.login_id)
    TextView mLoginID;


    private LoginPresenter loginPresenter;


    @Override
    protected int getLayout() {
        return R.layout.fragment_pass_word;
    }

    @Override
    protected void initView(View var1, Bundle var2) {


        String account = UserCenter.getInstance().getAccount();

        loginPresenter.attachView(this);

        if(!TextUtils.isEmpty(account)){

            mAccount.setText(account);
            mAccount.setSelection(account.length());
            mSaveBtn.setChecked(true);

        }

        mAccount.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                if(hasFocus){
                    mAccount.setSelected(true);
                }else {
                    mAccount.setSelected(false);
                }
            }
        });

        mAccount.setSelection(mAccount.getText().toString().length());
        mPassword.setSelection(mAccount.getText().toString().length());

//        mPassword.setOnFocusChangeListener(new View.OnFocusChangeListener() {
//            @Override
//            public void onFocusChange(View v, boolean hasFocus) {
//                if(hasFocus){
//                    mAccount.setSelected(true);
//                }else {
//                    mAccount.setSelected(false);
//                }
//            }
//        });

        mClickLayout.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                if(hasFocus){
                    mClickLayout.setBackgroundResource(R.drawable.border_color);
                }else {
                    mClickLayout.setBackgroundResource(android.R.color.transparent);
                }
            }
        });

        mLoginID.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                if(hasFocus){
                    mLoginID.setBackgroundResource(R.drawable.border_color);
                }else {
                    mLoginID.setBackgroundResource(android.R.color.transparent);
                }
            }
        });


        mSaveBtn.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {

                if(isChecked){
                    saveaccount = true;
                }else {
                    saveaccount = false;
                }
            }
        });

    }

    @Override
    protected void initData() {

    }

    @Override
    protected void initPresenter() {
        loginPresenter = new LoginPresenter(mContext);
        loginPresenter.attachView(this);
        Log.e("xx",""+loginPresenter);
    }


    @OnClick({R.id.btn_login_password,R.id.login_id,R.id.click_layout})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.btn_login_password:
                login();
                break;
            case R.id.login_id:
                LoginActivity loginActivity = (LoginActivity) getActivity();
                loginActivity.changeFragment(2);
                break;
            case R.id.click_layout:

                    if(!saveaccount){

                        saveaccount = true;
                        mSaveBtn.setChecked(true);
                        mSaveText.setTextColor(ContextCompat.getColor(mContext,R.color.text_blue));


                    }else {

                        saveaccount = false;
                        mSaveBtn.setChecked(false);
                        mSaveText.setTextColor(ContextCompat.getColor(mContext,R.color.white));
                        UserCenter.getInstance().delectUserid();

                    }

                break;
            default:
                break;
        }
    }


    private void login() {

        String mobile = mAccount.getText().toString().trim();
        String password = mPassword.getText().toString().trim();

        if (EmptyUtils.isEmpty(mobile)) {
            showMessageDialog("请输入手机号");
            return;
        }
        else if (!RegexUtils.isMobileExact(mobile)) {
            showMessageDialog("请输入正确的手机号");
            return;
        }

        if (EmptyUtils.isEmpty(password)) {
            showMessageDialog("请输入密码");
            return;
        }

        if(NetworkUtils.isConnected()) {

            loginPresenter.login(mobile, password, "SerialNo", 1,
                    EncryptUtils.encryptMD5ToString(mobile + password + "UNIS").toLowerCase());

        }else {

            Toast.makeText(mContext, "网络异常,请检查网络", Toast.LENGTH_SHORT).show();


        }
    }


    @Override
    public void onSuccess(LoginEntity result) {

        if (saveaccount) {
            UserCenter.getInstance().setUserid(mAccount.getText().toString().trim());
        }

        Intent intent = new Intent(mContext, MainActivity.class);

        intent.putExtra("username", result.getName());
        intent.putExtra("userhead", result.getHead());
        startActivity(intent);

        mContext.finish();


    }

    @Override
    public void showLoading() {

    }

    @Override
    public void hideLoading() {

    }

    @Override
    public void onError(Throwable throwable) {

    }


}

File: kotlin/src/main/java/com/unis/kotlin/ui/PassWordFragment.kt
package com.unis.kotlin.ui

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import android.widget.EditText
import android.widget.Toast
import com.blankj.utilcode.util.EmptyUtils
import com.blankj.utilcode.util.EncryptUtils
import com.blankj.utilcode.util.NetworkUtils
import com.blankj.utilcode.util.RegexUtils
import com.unis.kotlin.BaseApplication
import com.unis.kotlin.R
import com.unis.kotlin.base.BaseFragment
import com.unis.kotlin.cache.UserCenter
import com.unis.kotlin.constant.LoginConstant
import com.unis.kotlin.entity.LoginEntity
import com.unis.kotlin.presenter.LoginPresenter
import kotlinx.android.synthetic.main.fragment_password_login.*


class PassWordFragment : BaseFragment<LoginPresenter>() ,LoginConstant.View,View.OnClickListener{

    private var saveAccount : Boolean? = false

    private var loginPresenter : LoginPresenter? = null


    override fun getLayout(): Int {
        return R.layout.fragment_password_login
    }

    override fun initPresenter() {

    }



    override fun initView(var1: View, var2: Bundle?) {

        btn_login_password.setOnClickListener(this)
        login_id.setOnClickListener(this)

    }

    override fun initData() {




        txt_account.setSelection(txt_account.getText().toString().length)
        txt_password.setSelection(txt_account.getText().toString().length)



        click_layout.setOnFocusChangeListener(View.OnFocusChangeListener { view, hasFocus ->
            if (hasFocus) {
                click_layout.setBackgroundResource(R.drawable.border_color)
            } else {
                click_layout.setBackgroundResource(android.R.color.transparent)
            }
        })

        login_id.setOnFocusChangeListener(View.OnFocusChangeListener { v, hasFocus ->
            if (hasFocus) {
                login_id.setBackgroundResource(R.drawable.border_color)
            } else {
                login_id.setBackgroundResource(android.R.color.transparent)
            }
        })


        save_account.setOnCheckedChangeListener(CompoundButton.OnCheckedChangeListener { compoundButton, isChecked ->
            if (isChecked) {
                saveAccount = true
            } else {
                saveAccount = false
                UserCenter.get()!!.delectUserid()
            }
        })


    }



    override fun onClick(view: View?) {
        when(view?.id){
            R.id.btn_login_password ->{
                login()
            }
            R.id.login_id ->{

                val loginActivity = activity as LoginActivity?
                loginActivity!!.changeFragment(2)

            }

            R.id.click_layout ->{

                if (!saveAccount!!) {

                    saveAccount = true
                    save_account.setChecked(true)
                    save_text.setTextColor(ContextCompat.getColor(BaseApplication.getContext()!!, R.color.text_blue))



                } else {

                    saveAccount = false

                    save_account.setChecked(false)
                    save_text.setTextColor(ContextCompat.getColor(BaseApplication.getContext()!!, R.color.white))
                    UserCenter.get()!!.delectUserid()

                }

            }

        }


    }

    fun login(){

        val mobile = txt_account.getText().toString().trim()
        val password = txt_password.getText().toString().trim({ it <= ' ' })

        if (EmptyUtils.isEmpty(mobile)) {
            showMessageDialog("请输入手机号")
            return
        } else if (!RegexUtils.isMobileExact(mobile)) {
            showMessageDialog("请输入正确的手机号")
            return
        }

        if (EmptyUtils.isEmpty(password)) {
            showMessageDialog("请输入密码")
            return
        }

        if (NetworkUtils.isConnected()) {

            loginPresenter!!.login(mobile, password, "SerialNo", 1,
                    EncryptUtils.encryptMD5ToString(mobile + password + "UNIS").toLowerCase())

        } else {

            Toast.makeText(BaseApplication.getContext(), "网络异常,请检查网络", Toast.LENGTH_SHORT).show()


        }


    }


    override fun onDetach() {
        super.onDetach()
    }

    override fun onSuccess(result: LoginEntity) {
        super.onSuccess(result)

    }





}
